src/main/java/com/thealgorithms/datastructures/graphs/BoruvkaAlgorithm.java

```diff
@@ -0,0 +1,150 @@
+package com.thealgorithms.datastructures.graphs;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Boruvka's algorithm to find Minimum Spanning Tree
+ * (https://en.wikipedia.org/wiki/Bor%C5%AFvka%27s_algorithm)
+ *
+ * @author itakurah (https://github.com/itakurah)
+ */
+
+final class BoruvkaAlgorithm {
+    private BoruvkaAlgorithm() {
+    }
+    /**
+     * Represents an edge in the graph
+     */
+    static class Edge {
+        final int src;
+        final int dest;
+        final int weight;
+
+        Edge(int src, int dest, int weight) {
+            this.src = src;
+            this.dest = dest;
+            this.weight = weight;
+        }
+    }
+
+    /**
+     * Represents the graph
+     */
+    static class Graph {
+        final int vertex;
+        final List<Edge> edges;
+
+        /**
+         * Constructor for the graph
+         *
+         * @param vertex number of vertices
+         * @param edges list of edges
+         */
+        Graph(int vertex, List<Edge> edges) {
+            this.vertex = vertex;
+            this.edges = edges;
+        }
```

Comment:
Is there some connection between `vertex` and `edges`? Can they be completely arbitrary? i would add some `assert`'s.

Reply:
@vil02 added test cases for vertices and edges.